foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/platform/database/DB2ZPlatform.java

```diff
@@ -50,4 +82,394 @@ public String getTableCreationSuffix() {
+        int size = call.getParameters().size();
+        for (int index = indexFirst; index < size; index++) {
+            String name = call.getProcedureArgumentNames().get(index);
+            Object parameter = call.getParameters().get(index);
+            Integer parameterType = call.getParameterTypes().get(index);
+            // If the argument is optional and null, ignore it.
+            if (!call.hasOptionalArguments() || !call.getOptionalArguments().contains(parameter) || (row.get(parameter) != null)) {
+                if (name != null && shouldPrintStoredProcedureArgumentNameInCall()) {
+                    writer.write(getProcedureArgumentString());
+                    writer.write(name);
+                    writer.write(getProcedureArgumentSetter());
+                }
+                if (DatasourceCall.isOutputParameterType(parameterType)) {
+                    if (requiresProcedureCallOuputToken()) {
+                        writer.write(" ");
+                        writer.write(getOutputProcedureToken());
+                    }
+                }
+                if ((index + 1) < call.getParameters().size()) {
+                    writer.write(", ");
+                }
+            }
+        }
+
+        if (requiresProcedureCallBrackets()) {
+            writer.write(")");
+        }
+        writer.write(getProcedureCallTail());
+
+        return writer.toString();
+    }
+
+    /**
+     * This method is used to register output parameter on Callable Statements for Stored Procedures
+     * as each database seems to have a different method.
+     */
+    @Override
+    public void registerOutputParameter(CallableStatement statement, String name, int jdbcType, AbstractSession session) throws SQLException {
+        try {
+            Class clazz = null;
+            Method method = null;
+            String methodName = "registerJccOutParameterAtName";
+            Class[] methodArgs = new Class[] {String.class, int.class};
+            Object[] parameters = new Object[] {name, jdbcType};
+            if (PrivilegedAccessHelper.shouldUsePrivilegedAccess()) {
+                try {
+                    ClassLoader cl = AccessController.doPrivileged(new PrivilegedGetContextClassLoader(Thread.currentThread()));
+                    clazz = AccessController.doPrivileged(new PrivilegedClassForName(DB2_CALLABLESTATEMENT_CLASS, true, cl));
+                    method = AccessController.doPrivileged(new PrivilegedGetMethod(clazz, methodName, methodArgs, true));
+                    Object o = statement.unwrap(clazz);
+                    AccessController.doPrivileged(new PrivilegedMethodInvoker(method, o, parameters));
+                } catch (PrivilegedActionException ex) {
+                    if (ex.getCause() instanceof ClassNotFoundException) {
+                        throw (ClassNotFoundException) ex.getCause();
+                    }
+                    throw (RuntimeException) ex.getCause();
```

Comment:
Not sure if I'm a fan of just rethrowing a RuntimeException -- the caller might not only be expecting to handle SQLExceptions.  Yet just wrapping it within a SQLException isn't the answer either.